src/backend/distributed/commands/multi_copy.c

```diff
@@ -1421,22 +1684,31 @@ AppendCopyBinaryFooters(CopyOutState footerOutputState)
  */
 static void
 StartCopyToNewShard(ShardConnections *shardConnections, Oid relationId,
-					CopyStmt *copyStatement)
+					CopyStmt *copyStatement, PGconn *masterConnection)
 {
 	char *relationName = get_rel_name(relationId);
-	text *relationNameText = cstring_to_text(relationName);
-	Datum relationNameDatum = PointerGetDatum(relationNameText);
-	Datum shardIdDatum = DirectFunctionCall1(master_create_empty_shard,
-											 relationNameDatum);
+	int64 shardId = 0;
+
+	if (masterConnection == NULL)
```

Comment:
I think it looks slightly nicer if these if-else blocks are in their own function.